--- zm-1.21.0/web/zm_config.php.z.orig	2005-03-24 16:43:58.879820353 +0300
+++ zm-1.21.0/web/zm_config.php.z	2005-03-24 16:44:27.848615753 +0300
@@ -32,7 +32,7 @@
 	$str = fgets( $cfg, 256 );
 	if ( preg_match( '/^\s*$/', $str )) { continue; }
 	elseif ( preg_match( '/^\s*#/', $str )) { continue; }
-	elseif ( preg_match( '/^\s*([^=\s]+)\s*=\s*([^=\s]+)\s*$/', $str, $matches ))
+	elseif ( preg_match( '/^\s*([^=\s]+)\s*=\s*(\S+)\s*$/', $str, $matches ))
 	{
 		if (( $matches[1] == "ZM_DB_SERVER") ||
 			( $matches[1] == "ZM_DB_NAME") ||
