--- zm-1.21.0/src/zm_db.cpp.mysql41	2005-02-24 17:40:00.000000000 +0300
+++ zm-1.21.0/src/zm_db.cpp	2005-03-24 16:15:38.241465961 +0300
@@ -31,6 +31,16 @@
 		Error(( "Can't initialise structure: %s", mysql_error( &dbconn ) ));
 		exit( mysql_errno( &dbconn ) );
 	}
+#if MYSQL_VERSION_ID >= 40100
+	if ( mysql_options( &dbconn, MYSQL_READ_DEFAULT_GROUP, "client" ) )
+	{
+		Error(( "Unknown option MYSQL_READ_DEFAULT_GROUP" ));
+	}
+	if ( mysql_options( &dbconn, MYSQL_READ_DEFAULT_GROUP, "zoneminder" ) )
+	{
+		Error(( "Unknown option MYSQL_READ_DEFAULT_GROUP" ));
+	}
+#endif
 	if ( !mysql_real_connect( &dbconn, ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS, 0, 0, 0, 0 ) )
 	{
 		Error(( "Can't connect to server: %s", mysql_error( &dbconn ) ));
--- zm-1.21.0/scripts/zmaudit.pl.z.orig	2005-03-24 16:24:31.258199575 +0300
+++ zm-1.21.0/scripts/zmaudit.pl.z	2005-03-24 16:31:44.753311930 +0300
@@ -64,7 +64,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute: ".$sth->errstr() );
@@ -161,7 +161,7 @@
 	usage();
 }
 
-my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 chdir( EVENT_PATH );
 if ( $delay ) # Background mode
--- zm-1.21.0/scripts/zmcontrol-kx-hcm10.pl.z.orig	2005-03-24 16:25:38.995375323 +0300
+++ zm-1.21.0/scripts/zmcontrol-kx-hcm10.pl.z	2005-03-24 16:26:02.984895752 +0300
@@ -59,7 +59,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute '$sql': ".$sth->errstr() );
--- zm-1.21.0/scripts/zmcontrol-pelco-d.pl.z.orig	2005-03-24 16:26:08.557087511 +0300
+++ zm-1.21.0/scripts/zmcontrol-pelco-d.pl.z	2005-03-24 16:26:21.594196459 +0300
@@ -58,7 +58,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute '$sql': ".$sth->errstr() );
--- zm-1.21.0/scripts/zmcontrol-visca.pl.z.orig	2005-03-24 16:26:28.433204436 +0300
+++ zm-1.21.0/scripts/zmcontrol-visca.pl.z	2005-03-24 16:26:40.649432400 +0300
@@ -58,7 +58,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute '$sql': ".$sth->errstr() );
--- zm-1.21.0/scripts/zmdc.pl.z.orig	2005-03-24 16:26:50.413016111 +0300
+++ zm-1.21.0/scripts/zmdc.pl.z	2005-03-24 16:32:09.798680945 +0300
@@ -63,7 +63,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute: ".$sth->errstr() );
--- zm-1.21.0/scripts/zmfilter.pl.z.orig	2005-03-24 16:27:07.463542739 +0300
+++ zm-1.21.0/scripts/zmfilter.pl.z	2005-03-24 16:28:02.755521552 +0300
@@ -61,7 +61,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute '$sql': ".$sth->errstr() );
@@ -191,7 +191,7 @@
 
 chdir( EVENT_PATH );
 
-my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 print( "Scanning for events\n" );
 
--- zm-1.21.0/scripts/zmpkg.pl.z.orig	2005-03-24 16:27:25.732892487 +0300
+++ zm-1.21.0/scripts/zmpkg.pl.z	2005-03-24 16:27:43.762276972 +0300
@@ -62,7 +62,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute: ".$sth->errstr() );
@@ -93,7 +93,7 @@
 
 my $state;
 
-my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 if ( !$command || $command !~ /^(?:start|stop|restart|status)$/ )
 {
--- zm-1.21.0/scripts/zmtrack.pl.z.orig	2005-03-24 16:28:09.010614084 +0300
+++ zm-1.21.0/scripts/zmtrack.pl.z	2005-03-24 16:28:31.457357516 +0300
@@ -60,7 +60,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute '$sql': ".$sth->errstr() );
@@ -124,7 +124,7 @@
 
 print( "Tracker daemon $mid (experimental) starting at ".strftime( '%y/%m/%d %H:%M:%S', localtime() )."\n" );
 
-my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 my $sql = "select C.*,M.* from Monitors as M left join Controls as C on M.ControlId = C.Id where M.Id = ?";
 my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
--- zm-1.21.0/scripts/zmtrigger.pl.z.orig	2005-03-24 16:28:41.613883969 +0300
+++ zm-1.21.0/scripts/zmtrigger.pl.z	2005-03-24 16:29:21.387113303 +0300
@@ -61,7 +61,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute '$sql': ".$sth->errstr() );
@@ -119,7 +119,7 @@
 
 print( "Trigger daemon starting at ".strftime( '%y/%m/%d %H:%M:%S', localtime() )."\n" );
 
-my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 my $sql = "select * from Monitors where Id = ? or Name = ?";
 my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
--- zm-1.21.0/scripts/zmupdate.pl.z.orig	2005-03-24 16:29:27.171274056 +0300
+++ zm-1.21.0/scripts/zmupdate.pl.z	2005-03-24 16:30:09.367151455 +0300
@@ -69,7 +69,7 @@
 	}
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute: ".$sth->errstr() );
@@ -155,7 +155,7 @@
 
 if ( $check && ZM_CHECK_FOR_UPDATES )
 {
-	my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 	my $curr_version = ZM_DYN_CURR_VERSION;
 	my $last_version = ZM_DYN_LAST_VERSION;
@@ -193,7 +193,7 @@
 
 				print( "Got version: '".$last_version."'\n" );
 
-				my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+				my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 				my $lv_sql = "update Config set Value = ? where Name = 'ZM_DYN_LAST_VERSION'";
 				my $lv_sth = $dbh->prepare_cached( $lv_sql ) or die( "Can't prepare '$lv_sql': ".$dbh->errstr() );
@@ -246,7 +246,7 @@
 {
 	require DBI;
 
-	my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 	my $sql = "select Z.*, M.Width as MonitorWidth, M.Height as MonitorHeight from Zones as Z inner join Monitors as M on Z.MonitorId = M.Id where Z.Units = 'Percent'";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
@@ -321,7 +321,7 @@
 		# Nothing yet
 	}
 
-	my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 	my $cascade = undef;
 	if ( $cascade || $version eq "1.19.0" )
--- zm-1.21.0/scripts/zmvideo.pl.z.orig	2005-03-24 16:30:15.439270361 +0300
+++ zm-1.21.0/scripts/zmvideo.pl.z	2005-03-24 16:30:31.553932009 +0300
@@ -60,7 +60,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute: ".$sth->errstr() );
@@ -158,7 +158,7 @@
 select( STDERR ); $| = 1;
 select( LOG ); $| = 1;
 
-my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 my @filters;
 my $sql = "select max(F.Delta)-min(F.Delta) as FullLength, E.*, M.Name as MonitorName, M.Width as MonitorWidth, M.Height as MonitorHeight, M.Palette from Frames as F inner join Events as E on F.EventId = E.Id inner join Monitors as M on E.MonitorId = M.Id where EventId = '$event_id' group by F.EventId";
--- zm-1.21.0/scripts/zmwatch.pl.z.orig	2005-03-24 16:30:36.733180451 +0300
+++ zm-1.21.0/scripts/zmwatch.pl.z	2005-03-24 16:30:59.074938383 +0300
@@ -60,7 +60,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute: ".$sth->errstr() );
@@ -110,7 +110,7 @@
 print( "Watchdog pausing for ".START_DELAY." seconds\n" );
 sleep( START_DELAY );
 
-my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER, ZM_DB_USER, ZM_DB_PASS );
+my $dbh = DBI->connect( "DBI:mysql:database=".ZM_DB_NAME.";host=".ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", ZM_DB_USER, ZM_DB_PASS );
 
 my $sql = "select * from Monitors";
 my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
--- zm-1.21.0/scripts/zmx10.pl.z.orig	2005-03-24 16:31:06.512859022 +0300
+++ zm-1.21.0/scripts/zmx10.pl.z	2005-03-24 16:31:25.015173990 +0300
@@ -59,7 +59,7 @@
 	close( CONFIG );
 
 	use DBI;
-	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER, &ZM_DB_USER, &ZM_DB_PASS );
+	my $dbh = DBI->connect( "DBI:mysql:database=".&ZM_DB_NAME.";host=".&ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", &ZM_DB_USER, &ZM_DB_PASS );
 	my $sql = "select * from Config";
 	my $sth = $dbh->prepare_cached( $sql ) or die( "Can't prepare '$sql': ".$dbh->errstr() );
 	my $res = $sth->execute() or die( "Can't execute: ".$sth->errstr() );
@@ -207,7 +207,7 @@
 	bind( SERVER, $saddr ) or die( "Can't bind: $!" );
 	listen( SERVER, SOMAXCONN ) or die( "Can't listen: $!" );
 
-	$dbh = DBI->connect( "DBI:mysql:database=".main::ZM_DB_NAME.";host=".main::ZM_DB_SERVER, main::ZM_DB_USER, main::ZM_DB_PASS );
+	$dbh = DBI->connect( "DBI:mysql:database=".main::ZM_DB_NAME.";host=".main::ZM_DB_SERVER.";mysql_read_default_group=client;mysql_read_default_group=zoneminder", main::ZM_DB_USER, main::ZM_DB_PASS );
 
 	$x10 = new X10::ActiveHome( port=>main::ZM_X10_DEVICE, house_code=>main::ZM_X10_HOUSE_CODE, debug=>1 );
 
--- zm-1.21.0/web/zm_lang_pl_pl.php.orig	2005-03-24 17:58:00.556431008 +0300
+++ zm-1.21.0/web/zm_lang_pl_pl.php	2005-03-24 17:59:04.438007225 +0300
@@ -52,6 +52,17 @@
 // Example
 header( "Content-Type: text/html; charset=iso-8859-2" );
 
+if(version_compare(phpversion(), "4.3.10", ">=")>0) {
+    $result = mysql_query("SHOW VARIABLES LIKE 'version'");
+    $row = mysql_fetch_row($result);
+    $mysql_ver = $row[1];
+    if(version_compare($mysql_ver, '4.1.0', '>=')>0) {
+	mysql_query("set character set latin2");
+	mysql_query("set collation_connection='latin2_general_ci'");
+	mysql_query("set names latin2");
+    }
+}
+
 // You may need to change your locale here if your default one is incorrect for the
 // language described in this file, or if you have multiple languages supported.
 // If you do need to change your locale, be aware that the format of this function
--- zm-1.21.0/web/zm_lang_ru_ru.php.orig	2005-03-24 17:23:32.231404099 +0300
+++ zm-1.21.0/web/zm_lang_ru_ru.php	2005-03-24 17:56:29.804768214 +0300
@@ -52,6 +52,17 @@
 // Example
 header( "Content-Type: text/html; charset=koi8-r" );
 
+if(version_compare(phpversion(), "4.3.10", ">=")>0) {
+    $result = mysql_query("SHOW VARIABLES LIKE 'version'");
+    $row = mysql_fetch_row($result);
+    $mysql_ver = $row[1];
+    if(version_compare($mysql_ver, '4.1.0', '>=')>0) {
+	mysql_query("set character set koi8r");
+	mysql_query("set collation_connection='koi8r_general_ci'");
+	mysql_query("set names koi8r");
+    }
+}
+
 // You may need to change your locale here if your default one is incorrect for the
 // language described in this file, or if you have multiple languages supported.
 // If you do need to change your locale, be aware that the format of this function
