diff -urN xlib_shm-0.6.3.org/xlib_shm.c xlib_shm-0.6.3/xlib_shm.c
--- xlib_shm-0.6.3.org/xlib_shm.c	2006-12-06 13:42:56.000000000 +0100
+++ xlib_shm-0.6.3/xlib_shm.c	2007-12-04 08:43:30.414371892 +0100
@@ -48,7 +48,7 @@
 
 */
 #define VERSION "0.6.3"
-#define ZM_CONFIG "/usr/local/etc/zm.conf"
+#define ZM_CONFIG "/etc/zm.conf"
 
 
 #include <stdio.h>
@@ -226,7 +226,8 @@
     int r;
     if ( (cfg = fopen( ZM_CONFIG, "r")) == NULL )
     {
-	fprintf(stderr,"Can't open %s",ZM_CONFIG);
+	fprintf(stderr,"Can't open %s: %m",ZM_CONFIG);
+	exit(1);
     }
     while ( fgets( line, sizeof(line), cfg ) != NULL )
     {
